src/main.ts

```diff
@@ -97,16 +97,19 @@ async function postWithRetries(
       return result.data['accessToken'];
     } catch (error) {
       lastError = error;
-      if (isRetryableError(error) && attempt < maxRetries) {
-        continue;
+      if (isAxiosError(error)) {
+        const axiosErr: AxiosError = error as AxiosError;
+        if (isRetryableError(axiosErr) && attempt < maxRetries) {
```

Comment:
If we already know it's an AxiosError, does it need to be cast? 

```suggestion
      if (isAxiosError(error)) {
        if (isRetryableError(error) && attempt < maxRetries) {
```

Reply:
Very good point. Thank you.